scribus/ui/fontcombo.cpp

```diff
@@ -320,3 +322,101 @@ void FontComboH::RebuildList(ScribusDoc *currentDoc, bool forAnnotation, bool fo
 	fontFamily->blockSignals(familySigBlocked);
 	fontStyle->blockSignals(styleSigBlocked);
 }
+
+fontFamilyDelegate::fontFamilyDelegate(QObject *parent):
+	QAbstractItemDelegate(parent)
+{
+}
+
+void fontFamilyDelegate::paint(QPainter *painter,
+							   const QStyleOptionViewItem &option,
+							   const QModelIndex &index) const
+{
+	QString text = index.data(Qt::DisplayRole).toString();
+	QFont font;
+	QStringList families = QFontDatabase().families();
+	QString name, style;
+	for (int i =0; i < families.size(); ++i)
```

Comment:
We really need a more robust way to do this. Too many assumptions are being made.